Undefined `micros`

Some code crashes when attempting to use `micros()` because the linker generates a null value.

This bug was introduced by #1835. Adding `-u millis -u micros` to the linker fixes this problem, but then travis fails to build `HttpServer_ConfigNetwork` with SDK version 3:

```
out/Esp8266/debug/lib/libmain-rboot.a(mem_manager.o): In function `malloc':
(.text.malloc+0x8): multiple definition of `malloc'
/home/travis/build/SmingHub/Sming/Sming/Arch/Esp8266/Compiler/lib/libmicroc.a(malloc.o):C:\DEV\ESP8266\esp_microc/malloc/malloc.c:8: first defined here
```
This repeats for free, zalloc, calloc and realloc.

Clearly there is an underlying instability in how this is all linked. I'm looking to find a proper solution so the build is more robust, but any ideas most welcome.

One point I'm not clear on is with all the different SDK versions precisely which functions we're aiming to get linked. These are the problem areas:

`axtls-8266` implements the `time.c` module with these WEAK functions:

	configTime
	clock_gettime
	mktime
	time
	asctime
	ctime
	gettimeofday

(I took the `millis` and `micros` definitions out of the equation in #1835.)

Sming has `millis` and `micros` declared weak in `Clock.h`. If this is not done, we get these errors:

```
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `millis':
(.irom0.text+0xb4): undefined reference to `sntp_set_update_delay'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `millis':
(.irom0.text+0xbd): undefined reference to `sntp_set_update_delay'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `micros':
(.irom0.text+0xe0): undefined reference to `sntp_set_daylight'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `micros':
(.irom0.text+0xe4): undefined reference to `sntp_set_timetype'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `updateTime':
(.irom0.text+0xe8): undefined reference to `sntp_set_receive_time_size'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `updateTime':
(.irom0.text+0x134): undefined reference to `sntp_set_daylight'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `configTime':
(.irom0.text+0x13c): undefined reference to `sntp_set_timetype'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `configTime':
(.irom0.text+0x169): undefined reference to `sntp_set_receive_time_size'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `configTime':
(.irom0.text+0x184): undefined reference to `sntp_get_timetype'
out/Esp8266/debug/lib/libmain-rboot.a(time.o): In function `configTime':
(.irom0.text+0x197): undefined reference to `sntp_get_timetype'
```

--- a/Sming/Components/rboot/component.mk
+++ b/Sming/Components/rboot/component.mk
@@ -119,6 +119,7 @@ COMPONENT_LDFLAGS		:= -u Cache_Read_Enable_New
 $(LIBMAIN_RBOOT_DST): $(LIBMAIN_SRC)
 	@echo "OC $@"
 	$(Q) $(OBJCOPY) -W Cache_Read_Enable_New $^ $@
+	$(Q) $(AR) d $@ time.o
 
 ifeq ($(RBOOT_BIG_FLASH),1)
 LIBMAIN					:= $(LIBMAIN_RBOOT)

--- a/Sming/Core/Clock.h
+++ b/Sming/Core/Clock.h
@@ -29,7 +29,7 @@ extern "C" {
  *  @note   This function uses ESP8266 _system time_ clock which pauses during sleep. Function is provided for compatibility with Arduino. For date and time functionality, use SystemClock
  *  @see    SystemClockClass
  */
-unsigned long millis(void) WEAK_ATTR;
+unsigned long millis(void);
 
 /** @brief  Get the time from clock in microseconds
  *  @retval "unsigned long" Quantity of microseconds elapsed since clock epoch
@@ -37,7 +37,7 @@ unsigned long millis(void) WEAK_ATTR;
  *  @note   This function uses ESP8266 _system time_ clock which pauses during sleep. Function is provided for compatibility with Arduino. For date and time functionality, use SystemClock
  *  @see    SystemClockClass
  */
-unsigned long micros(void) WEAK_ATTR;
+unsigned long micros(void);
 
 /** @brief  Pause execution
  *  @param  milliseconds Duration of delay

